src/Security/InteractiveSecurityHelper.php

```diff
@@ -196,6 +180,96 @@ public function guessIdGetter(SymfonyStyle $io, string $userClass): string
         );
     }
 
+    /**
+     * @param array<string, array<string, mixed>> $firewalls Config data from security.firewalls
+     *
+     * @return Authenticator[]
+     */
+    public function getAuthenticatorsFromConfig(array $firewalls): array
+    {
+        $authenticators = [];
+
+        /* Iterate over each firewall that exists e.g. security.firewalls.main
+         * $firewallName could be "main" or "dev", etc...
+         * $firewallConfig should be an array of the firewalls params
+         */
+        foreach ($firewalls as $firewallName => $firewallConfig) {
+            if (!\is_array($firewallConfig)) {
+                continue;
+            }
+
+            $authenticators = [
+                ...$authenticators,
+                ...$this->getAuthenticatorsFromConfigData($firewallConfig, $firewallName),
+            ];
+        }
+
+        return $authenticators;
+    }
+
+    /**
+     * Pass in a firewalls config e.g. security.firewalls.main like:
+     *      pattern: ^/path
+     *      form_login:
+     *          login_path: app_login
+     *      custom_authenticator:
+     *          - App\Security\MyAuthenticator
+     *
+     * @param array<string, mixed> $firewallConfig
+     *
+     * @return Authenticator[]
+     */
+    private function getAuthenticatorsFromConfigData(array $firewallConfig, string $firewallName): array
+    {
+        $authenticators = [];
+
+        foreach ($firewallConfig as $potentialAuthenticator => $configData) {
```

Comment:
ahh... lost the review comment on push...

@weaverryan:
I'm actually a bit lost in this method. We're looping over the keys under a firewall, right? And the first if statement with tryFrom will exit unless the key matches the specific set we have in AuthenticatorType, right? So then, $configData is represents the values beneath those keys. So if I have:
```yaml
firewalls:
    main:
        form_login:
            path: /login
```

Won't $configData at this point be ['path' => '/login']? Or am I totally misreading the method?

Reply:
no I think you're thinking 1 level too deep..

We pass in `security.firewalls` -> iterate over each key in `firewalls` to give us:
```yaml
main:
    path: '/some-path'
    form_login:
        - option
        ....
    custom_authenticators:
        - some authenticator    
anotherFirewall:
dev:
test:
```

`tryFrom($potentialAuthenticator)` checks if the "key" is `form_login`, 'custom_authenticator`, or `path`. If it _is_ path, `tryFrom()` returns `null` and we `continue`.

Now `$potentialAuthenticator` is either `form_login` or `custom_authenticator` and `$configData` is the array of either of their stuff..

If its `form_login`, we create the value object and add it to `authenticators[]` otherwiser we now know that `$potentialAuthenticator === 'custom_authenticator'` and `configData` is either `(string) MyCustom` or [MyCustom, NotSureWhyYouCanHave2ButYouCan]

Reply:
Yup, it looks clear now 👍